backend/Ript-Fitness-Backend/src/main/java/com/riptFitness/Ript_Fitness_Backend/web/controller/UserProfileController.java

```diff
@@ -14,43 +13,58 @@
 import com.riptFitness.Ript_Fitness_Backend.infrastructure.service.UserProfileService;
 import com.riptFitness.Ript_Fitness_Backend.web.dto.UserDto;
 
+import jakarta.servlet.http.HttpServletRequest;
+
 @RestController 
 @RequestMapping("/userProfile") //base url for all userprofile endpoints
 public class UserProfileController {
-
-    // Each endpoint will call a method from UserProfileService
     private final UserProfileService userProfileService;
 
     // Constructor for dependency injection
     public UserProfileController(UserProfileService userProfileService) {
         this.userProfileService = userProfileService;
     }
 
-    //POST localhost:8080/userProfile/addUser
+    // POST localhost:8080/userProfile/addUser
     @PostMapping("/addUser")
     public ResponseEntity<UserDto> addUser(@RequestBody UserDto userDto) {
         UserDto savedUserObject = userProfileService.addUser(userDto);
         return new ResponseEntity<>(savedUserObject, HttpStatus.CREATED);
     }
 
-    //GET localhost:8080/userProfile/getUser/{userId}
-    @GetMapping("/getUser/{userId}")
-    public ResponseEntity<UserDto> getUser(@PathVariable Long userId) {
-        UserDto returnedUserObject = userProfileService.getUser(userId);
+    // GET localhost:8080/userProfile/getUserProfile
+    // Retrieves the profile of the logged-in user based on the username in the session
+    @GetMapping("/getUserProfile")
+    public ResponseEntity<UserDto> getUserProfile(HttpServletRequest request) {
+        String username = (String) request.getSession().getAttribute("username"); // Retrieve the username from session
```

Comment:
Can you move this to the Service class possibly? Just send the request object to the Service method and do the null check there?